//define syntax
syntax = "proto3";

package buyer;

import "seller.proto";

//define services
service BuyerService{

    //search item
    rpc SearchItem(SearchItemRequest) returns (SearchItemResponse);

    //buy item
    rpc BuyItem(BuyItemRequest) returns (BuyItemResponse);

    //add to wish list
    rpc AddToWishList(AddToWishListRequest) returns (AddToWishListResponse);

    //rate item
    rpc RateItem(RateItemRequest) returns (RateItemResponse);
}

//define request and response messages
message SearchItemRequest{
    string item_name = 1;
    string category_name = 2;
}

message SearchItemResponse{
    repeated seller.Item items = 1;
}

message BuyItemRequest{
    int32 item_id = 1;
    int32 item_quantity = 2;
}

message BuyItemResponse{
    enum Status{
        SUCCESS = 0;
        FAILURE = 1;
    }

    Status status = 1;
}

message AddToWishListRequest{
    int32 item_id = 1;
}

message AddToWishListResponse{
    enum Status{
        SUCCESS = 0;
        FAILURE = 1;
    }

    Status status = 1;
}

message RateItemRequest{
    int32 item_id = 1;
    int32 rating = 2;
}

message RateItemResponse{
    enum Status{
        SUCCESS = 0;
        FAILURE = 1;
    }

    Status status = 1;
}